src/System.Net.NameResolution/src/System/Net/NameResolutionPal.Unix.cs

```diff
@@ -110,134 +110,99 @@ private static unsafe IPHostEntry CreateHostEntry(Interop.libc.hostent* hostent)
 
         public static unsafe SocketError TryGetAddrInfo(string name, out IPHostEntry hostinfo, out int nativeErrorCode)
         {
-            var hints = new Interop.libc.addrinfo
+            Interop.Sys.HostEntry* entry = null;
+            int result = Interop.Sys.GetHostEntriesForName(name, &entry);
+            if (result != 0)
             {
-                ai_family = Interop.libc.AF_UNSPEC, // Get all address families
-                ai_flags = Interop.libc.AI_CANONNAME
-            };
+                hostinfo = NameResolutionUtilities.GetUnresolvedAnswer(name);
+                nativeErrorCode = result;
+                return GetSocketErrorForNativeError(result);
+            }
 
-            Interop.libc.addrinfo* addrinfo = null;
-            string canonicalName = null;
-            IPAddress[] ipAddresses;
             try
             {
-                int errorCode = Interop.libc.getaddrinfo(name, null, &hints, &addrinfo);
-                if (errorCode != 0)
-                {
-                    hostinfo = NameResolutionUtilities.GetUnresolvedAnswer(name);
-                    nativeErrorCode = errorCode;
-                    return GetSocketErrorForNativeError(errorCode);
-                }
+                string canonicalName = Marshal.PtrToStringAnsi((IntPtr)entry->CanonicalName);
 
-                int numAddresses = 0;
-                for (Interop.libc.addrinfo* ai = addrinfo; ai != null; ai = ai->ai_next)
+                hostinfo = new IPHostEntry
+                {
+                    HostName = string.IsNullOrEmpty(canonicalName) ? name : canonicalName,
+                    Aliases = Array.Empty<string>(),
+                    AddressList = new IPAddress[entry->Count]
+                };
+    
+                var buffer = new byte[SocketAddressPal.IPv6AddressSize];
+                for (int i = 0; i < entry->Count; i++)
                 {
-                    if (canonicalName == null && ai->ai_canonname != null)
+                    SocketAddress sockaddr;
+                    IPEndPoint factory;
+                    int bufferLength;
+                    if (entry->Addresses[i].IsIpv6)
                     {
-                        canonicalName = Marshal.PtrToStringAnsi((IntPtr)ai->ai_canonname);
-                    }
+                        sockaddr = new SocketAddress(AddressFamily.InterNetworkV6);
+                        factory = IPEndPointStatics.IPv6Any;
+                        bufferLength = SocketAddressPal.IPv6AddressSize;
```

Comment:
Please add a comment or file an issue to track updating this once `sockaddr` et. al. are shimmed out.